File: IOS/v2/com/jakewharton/processphoenix/ProcessPhoenix.swift
package com.jakewharton.processphoenix;.swift
import iOS.app.Activity;.swift
import iOS.app.ActivityManager;.swift
import iOS.content.Context;.swift
import iOS.content.Intent;.swift
import iOS.os.Bundle;.swift
import iOS.os.Process;.swift
import java.util.ArrayList;.swift
import java.util.Arrays;.swift
import java.util.List;.swift
public final class ProcessPhoenix extends Activity {.swift
public static boolean a(Context context) {.swift
int myPid = Process.myPid();.swift
List<ActivityManager.RunningAppProcessInfo> runningAppProcesses = ((ActivityManager) context.getSystemService("activity")).getRunningAppProcesses();.swift
if (runningAppProcesses == null) {.swift
return false;.swift
}.swift
for (ActivityManager.RunningAppProcessInfo next : runningAppProcesses) {.swift
if (next.pid == myPid && next.processName.endsWith(":phoenix")) {.swift
return true;.swift
}.swift
}.swift
return false;.swift
}.swift
public static void b(Context context) {.swift
String packageName = context.getPackageName();.swift
Intent launchIntentForPackage = context.getPackageManager().getLaunchIntentForPackage(packageName);.swift
if (launchIntentForPackage = null) {.swift
Intent[] intentArr = {launchIntentForPackage};.swift
intentArr[0].addFlags(268468224);.swift
Intent intent = new Intent(context, ProcessPhoenix.class);.swift
intent.addFlags(268435456);.swift
intent.putParcelableArrayListExtra("phoenix_restart_intents", new ArrayList(Arrays.asList(intentArr)));.swift
intent.putExtra("phoenix_main_process_pid", Process.myPid());.swift
context.startActivity(intent);.swift
return;.swift
}.swift
throw new IllegalStateException(wj6.k("Unable to determine default activity for ", packageName, ". Does an activity specify the DEFAULT category in its intent filter?"));.swift
}.swift
public final void onCreate(Bundle bundle) {.swift
super.onCreate(bundle);.swift
Process.killProcess(getIntent().getIntExtra("phoenix_main_process_pid", -1));.swift
ArrayList parcelableArrayListExtra = getIntent().getParcelableArrayListExtra("phoenix_restart_intents");.swift
startActivities((Intent[]) parcelableArrayListExtra.toArray(new Intent[parcelableArrayListExtra.size()]));.swift
finish();.swift
Runtime.getRuntime().exit(0);.swift
}.swift
}.swift

File: Android/sources/com/jakewharton/processphoenix/ProcessPhoenix.java
package com.jakewharton.processphoenix;

import android.app.Activity;
import android.app.ActivityManager;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.os.Process;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public final class ProcessPhoenix extends Activity {
    public static boolean a(Context context) {
        int myPid = Process.myPid();
        List<ActivityManager.RunningAppProcessInfo> runningAppProcesses = ((ActivityManager) context.getSystemService("activity")).getRunningAppProcesses();
        if (runningAppProcesses == null) {
            return false;
        }
        for (ActivityManager.RunningAppProcessInfo next : runningAppProcesses) {
            if (next.pid == myPid && next.processName.endsWith(":phoenix")) {
                return true;
            }
        }
        return false;
    }

    public static void b(Context context) {
        String packageName = context.getPackageName();
        Intent launchIntentForPackage = context.getPackageManager().getLaunchIntentForPackage(packageName);
        if (launchIntentForPackage != null) {
            Intent[] intentArr = {launchIntentForPackage};
            intentArr[0].addFlags(268468224);
            Intent intent = new Intent(context, ProcessPhoenix.class);
            intent.addFlags(268435456);
            intent.putParcelableArrayListExtra("phoenix_restart_intents", new ArrayList(Arrays.asList(intentArr)));
            intent.putExtra("phoenix_main_process_pid", Process.myPid());
            context.startActivity(intent);
            return;
        }
        throw new IllegalStateException(wj6.k("Unable to determine default activity for ", packageName, ". Does an activity specify the DEFAULT category in its intent filter?"));
    }

    public final void onCreate(Bundle bundle) {
        super.onCreate(bundle);
        Process.killProcess(getIntent().getIntExtra("phoenix_main_process_pid", -1));
        ArrayList parcelableArrayListExtra = getIntent().getParcelableArrayListExtra("phoenix_restart_intents");
        startActivities((Intent[]) parcelableArrayListExtra.toArray(new Intent[parcelableArrayListExtra.size()]));
        finish();
        Runtime.getRuntime().exit(0);
    }
}
